### PR TITLE
Add friendly error message for missing dotfiles

### LINKED ISSUE #544: TypeError: Cannot convert undefined or null to object in task lint
Download 1.2 directly, then did npm install and bower install, then do "gulp serve", and here is the error output:
```
[12:37:13] Starting 'lint'...
[12:37:14] 'lint' errored after 1.18 s
[12:37:14] TypeError: Cannot convert undefined or null to object
    at Function.keys (native)
    at copyConfiguration (/Users/cnbuff410/node_modules/gulp-jscs/node_modules/jscs/lib/config/configuration.js:1181:12)
    at NodeConfiguration.Configuration._processConfig (/Users/cnbuff410/node_modules/gulp-jscs/node_modules/jscs/lib/config/configuration.js:474:5)
    at NodeConfiguration.Configuration.load (/Users/cnbuff410/node_modules/gulp-jscs/node_modules/jscs/lib/config/configuration.js:238:10)
    at StringChecker.configure (/Users/cnbuff410/node_modules/gulp-jscs/node_modules/jscs/lib/string-checker.js:67:29)
    at Checker.configure (/Users/cnbuff410/node_modules/gulp-jscs/node_modules/jscs/lib/checker.js:28:39)
    at Object.module.exports (/Users/cnbuff410/node_modules/gulp-jscs/index.js:35:10)
    at Gulp.<anonymous> (/Users/cnbuff410/gulpfile.js:124:11)
    at module.exports (/Users/cnbuff410/node_modules/gulp/node_modules/orchestrator/lib/runTask.js:34:7)
    at Gulp.Orchestrator._runTask (/Users/cnbuff410/node_modules/gulp/node_modules/orchestrator/index.js:273:3)
```
It's caused by this line in lint task:

  .pipe($.jscs())

### PR DESCRIPTION
**Pull request by samccone**
via https://github.com/PolymerElements/polymer-starter-kit/pull/555

fixes #544

R @robdodson

### CODE REVIEW

**Comment by chuckh** [comment]
LGTM!

File: tasks/ensure-files.js [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+
+    if (!fileFound) {
+      prev.push(filePath + ' Not Found');
+    }
+
+    return prev;
+  }, []);
+
+  if (missingFiles.length) {
+    throw new Error('Missing Required Files\n' + missingFiles.join('\n'));
```
Comment (robdodson) [inline tasks/ensure-files.js]: In gulp land, is the preference to throw or return the error here?

Comment (samccone) [inline tasks/ensure-files.js]: hmm my through was to throw inside of the task and if we want to pass an error out we can do that in the gulpfile... let me tweak it as see if passing an error explicitly looks nicer

Comment (samccone) [inline tasks/ensure-files.js]: good call, it is better

**Comment by samccone** [comment]
best

day


ever

thanks a ton @ohanhi :cactus:

**Comment by ohanhi** [comment]
This is great! 🎈🎉🎊

Thank you, @samccone!

**Pull request merged**